thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -842,12 +843,33 @@ namespace diskann {
     const T *    query = data.scratch.aligned_query_T;
     const float *query_float = data.scratch.aligned_query_float;
 
-    for (uint32_t i = 0; i < this->data_dim; i++) {
+    auto q_dim = this->data_dim;
+    if(metric== diskann::Metric::INNER_PRODUCT && !smaple){ // query_dim need specify when using IP
+      q_dim--;
+    }
+    for (uint32_t i = 0; i < q_dim; i++) {
       data.scratch.aligned_query_float[i] = query1[i];
       data.scratch.aligned_query_T[i] = query1[i];
       query_norm += query1[i] * query1[i];
     }
 
+    if(query_norm <= 0 && metric == diskann::Metric::INNER_PRODUCT){
+    // random answer for zero point ip search 
```

Comment:
Why we do random answer

Reply:
> Why we do random answer

The distance between zero point and all points is the same, it is better to return a random answer